src/img/gen-color-table.pl

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/perl -w
+
+# Generates xpm-color-table.h from X11's rgb.txt
+# Adapted from gdk-pixbuf (LGPL-2.0-or-later)
+
+if (@ARGV != 1) {
+    die "Usage: gen-color-table.pl rgb.txt > xpm-color-table.h\n";
+}
+
+open IN, $ARGV[0] || die "Cannot open $ARGV[0]: $!\n";
+
+@colors = ();
+while (defined($_ = <IN>)) {
+    next if /^!/;
+    if (!/^\s*([0-9]+)\s+([0-9]+)\s+([0-9]+)\s+(.*\S)\s+$/) {
+	die "Cannot parse line $_";
+    }
+
+    push @colors, [$1, $2, $3, $4];
+}
+
+@colors = sort { lc($a->[3]) cmp lc($b->[3]) } @colors;
```

Comment:
hmm, lowercasing the lines for comparing (but keeping the original casing in @colors.

is the hash code (which the change in check hints) going to do case-insensitive searching....

... if so, one could also remove whitespace (and then remove duplicates) to shorten the list
(there could be duplicates even without doing this suggested whitespace removal..)

anyway that is just a thought that came in my mind, i don't know how good it is

Reply:
The lookup is a binary-search, ASCII case-insensitive (using `g_ascii_strcasecmp`). I suppose removing whitespace would also possible, but a little more complicated and I'd rather not mess with this too much.